scriptoocloud/src/test/java/edu/asu/diging/scriptoocloud/core/service/impl/JgitServiceImplTest.java

```diff
@@ -0,0 +1,50 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.File;
+import java.net.MalformedURLException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.scriptoocloud.core.data.GitRepositoryRepository;
+import edu.asu.diging.scriptoocloud.core.exceptions.InvalidGitUrlException;
+import edu.asu.diging.scriptoocloud.core.service.DeleteFilesService;
+
+class JgitServiceImplTest {
+
+    @InjectMocks
+    private JgitServiceImpl serviceToTest;
+    
+    @Mock
+    private DeleteFilesService deleteFilesService;
+    
+    @BeforeEach
+    public void init(){
+        MockitoAnnotations.initMocks(this);  
+    } 
+    
+    @Test
+    public void test_clone_success(){
+        Assertions.assertDoesNotThrow(()->serviceToTest.clone("testCloneSuccess", "https://github.com/diging/scriptoocloud"));
```

Comment:
I wouldn't use a relative folder here, but make that configurable. You should be able to create a properties object from a file in `src/test/resources` and then use the configured location here (similar to how it is done in the actual code). You also want to make sure your that you clean up after your tests (removing the created directory, so it doesn't interfere with your next test).

Reply:
Last but not least, does it make a difference if the folder where the repository should be cloned has contents? Since Git doesn't like it if you do it on the command line, I assume it does? In that case, you should have a test case for both scenarios.

Reply:
Can I remove this clone success test since it's just the standard JGit clone call? I've spent a lot of time trying to figure out how to kill the JGit clone process and release file system locks its holding and I'm not having any success. This results in my cleanup call in @AfterEach only partially cleaning up the folder created. If this test is necessary the only solution I can think of is modifying my JGit service to accommodate tests better?

Reply:
Let's talk about it in our meeting today, I think I don't understand the problems you are running into.